internal/pkg/vm/actor/actor.go

```diff
@@ -71,43 +68,18 @@ func (a *Actor) IncrementSeqNum() {
 	a.CallSeqNum = a.CallSeqNum + 1
 }
 
-// Cid returns the canonical CID for the actor.
-// TODO: can we avoid returning an error?
-func (a *Actor) Cid() (cid.Cid, error) {
-	bs, err := encoding.Encode(a)
-	if err != nil {
-		return cid.Undef, errors.Wrap(err, "failed to marshal to cbor")
-	}
-	obj, err := cbor.Decode(bs, types.DefaultHashFunction, -1)
-	if err != nil {
-		return cid.Undef, errors.Wrap(err, "failed to cast cbor bytes to ipld node")
-	}
-
-	return obj.Cid(), nil
-}
-
-// Unmarshal a actor from the given bytes.
-func (a *Actor) Unmarshal(b []byte) error {
-	return fxamackercbor.Unmarshal(b, a)
-}
-
-// Marshal the actor into bytes.
-func (a *Actor) Marshal() ([]byte, error) {
-	return fxamackercbor.Marshal(a, fxamackercbor.EncOptions{})
-}
-
 // UnmarshalCBOR must implement cbg.Unmarshaller to insert this into a hamt.
```

Comment:
This comment might be false after @ZenGround0's recent work on HAMT

Reply:
I tried to remove this during that HAMT work but specs actors adts used in the state view ended up depending on this too.  One day we should clean it up but its very low priority right now.

Reply:
Ok. It would be possible to change the state view to not use the ADT abstractions at the top level and hence not rely on this.